File: backend/src/main/kotlin/outcobra/server/model/dto/mark/SemesterMarkDto.kt
package outcobra.server.model.dto.mark

import outcobra.server.model.dto.ColorDto
import outcobra.server.model.dto.MarkGroupDto
import outcobra.server.model.dto.SchoolClassDto
import java.time.LocalDate

/**
 * @author Florian Bürgi
 * @since 1.2.0
 */
data class SemesterMarkDto(val id: Long = 0,
                           val name: String = "",
                           val validFrom: LocalDate = LocalDate.now(),
                           val validTo: LocalDate = LocalDate.now(),
                           val schoolClass: SchoolClassDto,
                           val value: Double = 0.0,
                           val subjects: List<SubjectMarkDto> = listOf())

/**
 * @author Florian Bürgi
 * @since 1.2.0
 */
data class SubjectMarkDto(val id: Long = 0,
                          val name: String = "",
                          val color: ColorDto = ColorDto(),
                          val subjectMarkGroup: MarkGroupDto = MarkGroupDto())

File: frontend/src/app/mark/model/semester-mark.dto.ts
import { Moment } from 'moment';
import { ColorDto } from '../../core/model/color.dto';
import { InstitutionDto, SchoolClassDto } from '../../manage/model/manage.dto';
import { MarkGroupDto } from './mark-group.dto';

export interface SemesterMarkDto {
  id: number,
  name: string,
  validFrom: Moment,
  validTo: Moment,
  institution: InstitutionDto,
  schoolClass: SchoolClassDto,
  value: number,
  subjects: Array<SubjectMarkDto>
}

export interface SubjectMarkDto {
  id: number,
  name: string,
  color: ColorDto,
  subjectMarkGroup: MarkGroupDto
}
